ui/src/app/shared/genericComponents/chart/chart.ts

```diff
@@ -10,14 +10,14 @@ import { Edge, Service } from "../../shared";
   selector: 'oe-chart',
   templateUrl: './chart.html'
 })
-export class ChartComponent implements OnInit, OnChanges {
+export class ChartComponent implements OnInit, AfterViewChecked {
 
   public edge: Edge | null = null;
   @Input() public title: string = '';
-  @Input() public showPhases: boolean;
-  @Input() public showTotal: boolean;
-  @Output() public setShowPhases: EventEmitter<boolean> = new EventEmitter();
-  @Output() public setShowTotal: EventEmitter<boolean> = new EventEmitter();
+  @Input() public showPhases: boolean | null = null;
+  @Input() public showTotal: boolean | null = null;
+  @Output() public showTotalChange: EventEmitter<boolean> = new EventEmitter();
+  @Output() public showPhasesChange: EventEmitter<boolean> = new EventEmitter();
   @Input() public isPopoverNeeded: boolean = false;
 
   // Manually trigger ChangeDetection through Inputchange
```

Comment:
shouldn't period be 'public'.

"So, while you could technically make an @Input property private, it's not a recommended practice because it goes against the intended use of the @Input decorator. The whole point of using @Input is to make a property accessible from outside the component."

Reply:
period?